website/docs/generics.md

```diff
@@ -0,0 +1,1019 @@
+---
+id: generics
+title: Generic Methods and Classes
+sidebar_label: Generics
+---
+
+Sorbet has syntax for creating generic methods, classes, and interfaces.
+
+## Implementation status
+
+Generics are, by far, the most poorly supported feature within Sorbet. The
+implementation has many known, hard-to-fix bugs, which substantially limits
+their utility.
+
+It is hard to overstate how buggy Sorbet's support for generics is.
+
+We track known bugs in Sorbet's support for generics in the [Generics milestone]
+on Sorbet's issue tracker. Feel free to peruse the existing bugs, as well as
+report new bugs.
+
+Unfortunately, the abundance of bugs in Sorbet's generics implementation makes
+implementing sophisticated, generic abstractions in Sorbet exceedingly brittle.
+
+It is imperative to thoroughly test abstractions making use of Sorbet generics.
+
+The tests you'll need to write look substantially different from other Ruby
+tests you may be accustomed to writing, because these tests will have to live
+**at the type level** (statics), not at the expression level (runtime). These
+tests should account for two main failure modes:
+
+- Many things that shouldn't type check **do type check anyways**.
+
+  This is bad because the generic abstraction being built will not necessarily
+  provide the guarantees it should. This can give users of the generic
+  abstraction false confidence in the type system.
+
+  To mitigate this, write tests using the abstraction that should not type
+  check, and double check that they don't. Get creative with these tests.
+  Consider writing tests that make uncommon use of subtyping, inheritance,
+  mutation, etc.
+
+  It can also be helpful to use `T.reveal_type` and/or `T.assert_type!` to
+  inspect the types of intermediate values to see if `T.untyped` has silently
+  snuck in somewhere.
+
+  (There is no method built into Sorbet for writing such tests; the best way to
+  do it is manually while making changes to the generic abstraction. The
+  diligent may way to automate this by running Sorbet a second time on a
+  codebase that includes extra files meant to not type check, and assert that
+  Sorbet indeed reports errors.)
+
+- Many things **don't type check when they should**.
+
+  This is bad because it will cause undue toil to people attempting to use the
```

Comment:
specially for readers that have English as a second language, "undue toil"  is more complicated than needed.

Is the result really "undue toil" (what are they going to do?) or "confusion"?